### PR TITLE
Deal with surrogate pairs.

### LINKED ISSUE #8: flip function results in mojibake for wide characters in narrow Python builds
The `flip` function produces mojibake if the input string contains wide characters, when using Xenogossia on narrow Python builds.

Expected (as seen in Python 3):

``` xg
$ echo 'test😌' | xg 'flip' | pbcopy
（╯°□°）╯︵ 😌ʇsǝʇ
```

Actual (as seen in a narrow build of Python 2.7):

``` xg
# output is completely broken when piping into another program
$ echo 'test😌' | xg 'flip' | pbcopy
Ôºà‚ïØ¬∞‚ñ°¬∞Ôºâ‚ïØÔ∏µ Ì∏åÌ†Ω ás«ù á
# results are easier to read when stdout has an encoding
$ echo 'test😌' | xg 'flip'
（╯°□°）╯︵ ������ʇsǝʇ
```

### PR DESCRIPTION
**Pull request by kivikakk**
(Also include PEP-0263 encoding line?)

Tested on python 2.7.8 and 3.4.3 (both on OS X).

Fixes #8.

### CODE REVIEW

File: xenoglossia/builtins.py [hunk 1]
```diff
@@ -167,7 +168,22 @@ def flip(input, *args):
     """
     Returns an upside down version of *input* prepended by our favorite Flip Table dude.
     """
-    flipped_input = "".join(_FLIP_CHAR_DICT.get(c, c) for c in input[::-1])
+    char_list = []
+    pair = None
+    for c in input[::-1]:
+        if ord(c) in range(0xDC00, 0xE000):
```
Comment (mgedmin) [inline xenoglossia/builtins.py:165]: This does a linear search in a materialized list on Python 2.  Suggest using `if 0xDC00 <= ord(c) < 0xE000:` instead.

Comment (kivikakk) [inline xenoglossia/builtins.py:165]: `range` is imported from `six.moves`, which points to `xrange` on Python 2.

Comment (mgedmin) [inline xenoglossia/builtins.py:165]: Ah, cool!

**Comment by mistydemeo** [comment]
Nice!! Thanks very much!

I wonder if it's worth pulling this out into a helper function in a new `utils` module, in case other functions need to do the same?

**Comment by kivikakk** [comment]
Maybe; I tend to go by the rule of three for this kind of thing.  We don't know exactly what other uses might look like, so maybe best to wait it out until they come up. (unless they're all literally wanting reversed text, but who knows)

**Comment by mistydemeo** [comment]
I was actually considering adding a direct `reverse` function to the builtins! Which could use the same logic as this when operating on strings.

**Comment by kivikakk** [comment]
'tis done!

**Comment by mistydemeo** [comment]
:sparkling_heart: Thanks so much!!

**Pull request closed without merging**